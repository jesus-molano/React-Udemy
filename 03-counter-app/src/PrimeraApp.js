import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo }) => {

  // const obj = {name: 'Jesus', age: 26,}

  return (
    <>
      <h1>{saludo}</h1>
      <h3>{subtitulo}</h3>
      {/* <pre>{JSON.stringify(obj, null, 2)}</pre> */}
    </>
  );
}
PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
  subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;