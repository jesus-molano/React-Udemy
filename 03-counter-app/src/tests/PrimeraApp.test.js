// import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme';

describe('<PrimeraApp /> tests', () => {
  // testing-library ---
  // test('should show the message "Hola, Soy Goku"', () => {
  //   const saludo = "Hola, Soy Goku";
  //   const { getByText } = render(<PrimeraApp saludo={saludo} />)

  //   expect(getByText(saludo)).toBeInTheDocument();
  // })

  test('should show PrimeraApp component correctly', () => {
    const saludo = 'Hola, Soy Goku'
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  })
  test('should show the prop => subtitle', () => {
    const saludo = 'Hola, Soy Goku'
    const subtitulo = "Soy un subtitulo"
    const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo} />);
    const textoSubtitulo = wrapper.find('h3').text();

    expect(textoSubtitulo).toBe(subtitulo);
  })

})
