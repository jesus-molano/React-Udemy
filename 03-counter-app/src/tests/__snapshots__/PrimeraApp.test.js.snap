// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimeraApp /> tests should show PrimeraApp component correctly 1`] = `
<Fragment>
  <h1>
    Hola, Soy Goku
  </h1>
  <h3>
    Soy un subtitulo
  </h3>
</Fragment>
`;
